holoviews/core/spaces.py

```diff
@@ -856,7 +890,8 @@ def __getitem__(self, key):
         try:
             dimensionless = util.dimensionless_contents(get_nested_streams(self),
                                                         self.kdims, no_duplicates=False)
-            if dimensionless:
+            empty = util.stream_parameters(self.streams) == [] and self.kdims==[]
```

Comment:
No streams or kdims implies it is empty i.e if valid, a callable without arguments or a generator.